engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -1027,6 +1027,19 @@ public int processEvents()
         }
       }
       o.stats.lastWindowedStats = stats;
+      if (!o.stats.responses.isEmpty()) {
+        if (o.stats.operatorResponses == null) {
+          o.stats.operatorResponses = new ArrayList<>();
+        } else {
+          o.stats.operatorResponses.clear();
+        }
+        StatsListener.OperatorResponse operatorResponse;
+        while ((operatorResponse = o.stats.responses.poll()) != null) {
+          o.stats.operatorResponses.add(operatorResponse);
+        }
+      } else {
+        o.stats.operatorResponses = null;
```

Comment:
@tweise : Initially it was done that way but @amberarrow suggested that we reuse the same list if possible, so made this change. I am fine with either way...